api/tests/validate_legacy_soap_api/validate_soap_endpoints.py

```diff
@@ -0,0 +1,183 @@
+import io
+import logging
+import os
+import re
+import tempfile
+import zipfile
+
+import click
+import requests
+from lxml import etree
+
+import src.adapters.db as db
+import src.logging
+import tests.src.db.models.factories as factories
+from src.adapters.db import PostgresDBClient
+from src.legacy_soap_api.legacy_soap_api_config import SimplerSoapAPI
+from src.legacy_soap_api.legacy_soap_api_schemas import SOAPRequest, SOAPResponse
+from src.legacy_soap_api.legacy_soap_api_utils import get_alternate_proxy_response
+from src.legacy_soap_api.simpler_soap_api import get_simpler_soap_response
+from src.util.local import error_if_not_local
+
+logger = logging.getLogger(__name__)
+
+
+"""
+Add following to soap-api.env
+CERT_DATA_ENCODED =
+KEY_DATA_ENCODED =
+SOAP_URI =
+For cert and key use awk command to replace newlines
+`awk 'NF {sub(/\r/, ""); printf "%s\\n",$0;} bps_grantors.crt`
+`awk 'NF {sub(/\r/, ""); printf "%s\\n",$0;} bps_grantors.key`
+May need to run `poetry install` to update command
+To run use: `docker compose run --rm grants-api poetry run validate-simpler-endpoints`
+"""
+CERT_DATA_ENCODED = os.getenv("CERT_DATA")
+KEY_DATA_ENCODED = os.getenv("KEY_DATA")
+SOAP_URI = os.getenv("SOAP_URI", b"")
+HEADERS = {"Content-Type": "application/xml"}
+
+if CERT_DATA_ENCODED and KEY_DATA_ENCODED:
+    CERT_DATA = CERT_DATA_ENCODED.replace("\\n", "\n")
+    KEY_DATA = KEY_DATA_ENCODED.replace("\\n", "\n")
+else:
+    raise Exception("No cert data")
+with tempfile.NamedTemporaryFile(delete=False) as temp_cert_file:
+    temp_cert_file.write(CERT_DATA.encode("utf-8"))
+    CERT = temp_cert_file.name
+with tempfile.NamedTemporaryFile(delete=False) as temp_key_file:
+    temp_key_file.write(KEY_DATA.encode("utf-8"))
+    KEY = temp_key_file.name
```

Comment:
I feel like we shouldn't be keeping temporary files with cert info around.

Could we move these to be in the main loop / put some of this in some sort of context we pass around? That way we can use a tempfile to make the file and let it delete it when we're done.

Reply:
Got to leverage ExitStack and I think it's working now that they're automatically deleted once you exit the context.